app/main.py

```diff
@@ -16,17 +16,12 @@ def is_list_of_strs(input):
 
     def prediction(self, input):
         if 'words' not in input:
-            raise KeyError("No key named 'words' in input.")
+            raise KeyError("Expected key named 'words' in input.")
         if not self.is_list_of_strs(input['words']):
             raise ValueError("'words' should be a list of strings.")
-
-        limit = DEFAULT_LIMIT
-        if 'limit' in input:
-            try:
-                limit = int(str(input['limit']))
-                limit = max(MIN_LIMIT, min(limit, MAX_LIMIT))
-            except ValueError:
-                raise ValueError("'limit' must be an integer in the range 1-100.")
+        limit = max(MIN_LIMIT, min(input['limit'], MAX_LIMIT)) if 'limit' in input else DEFAULT_LIMIT
+        if not isinstance(limit, int):
```

Comment:
Can we test this part?

Reply:
Also i think this need to be above the min/max logic. how can you do the min/max operation on a non-number?

Reply:
min/max logic should be below the int validation. We do have tests for that part but I should tweak the int test (instead of testing on a float test on str limit) or add an additional assertions to that test. 

https://github.com/DopplerFoundation/example-app-similar-words/blob/aad134861e9f6c009c392bb35c28274d4b802e06/tests.py#L38-L52